README.md

```diff
@@ -37,16 +37,166 @@ NOTE: we suggest using virtualenv to sandbox your setup.
 - [VALIDATOR_GUIDE.md](https://github.com/ethereum/casper/blob/master/VALIDATOR_GUIDE.md):
 information about implementing a Casper FFG validator.
 
-## Contribute
+## Tests
 
-Feel free to ask questions in our [Gitter room](https://gitter.im/ethereum/casper) or open an [issue](https://github.com/ethereum/casper/issues) for feature requests or bug reports. Feel free to make a PR!
+```bash
+pytest tests
+```
 
-## License
+## Version of Vyper to Compile Casper
+We recommend using the vyper version installed as a dependency from requirements.txt. 
+NOTE: The latest version of Vyper throws an error (line 16) when compiling the Casper contract due to a syntax change of "units".
+NOTE: this not needed if steps above followed
+```bash
+git clone https://github.com/ethereum/vyper.git@248c723288e84899908048efff4c3e0b12f0b3dc
+```
+The guide on how to install Vyper can be found [here](https://github.com/ethereum/vyper#installation).
 
-[UNLICENSE](LICENSE)
 
-## Tests
+## How to Build and Deploy Casper Contract
+
+In this example we will be using [eth-tester](https://github.com/ethereum/eth-tester).
+
 
+### Step 1: Imports
 ```bash
-pytest tests
+import os
+import rlp
+
+from decimal import (
+    Decimal,
+)
+
+import eth_tester
+
+from eth_tester import (
+    EthereumTester,
+    PyEVMBackend
+)
+from web3.providers.eth_tester import (
+    EthereumTesterProvider,
+)
+
+from web3 import (
+    Web3,
+)
+
+from vyper import (
+    compiler,
+    utils as vyper_utils,
+)
+
+```
+
+### Step 2: Raise the Gas Limit
+
+```bash
+setattr(eth_tester.backends.pyevm.main, 'GENESIS_GAS_LIMIT', 10**9) ## it costs a lot of gas to deploy the casper contract, so set a high gas limit
+```
+### Step 3: Create an Instance of Web3 (eth-tester)
+
+[Link to the web3.py documentation.](https://github.com/ethereum/web3.py)
+```bash
+base_tester = EthereumTester(PyEVMBackend())  ## instantiates the eth-test suite 
+w3 = Web3(EthereumTesterProvider(base_tester)) ## Create an instance of the web3 object
+w3.eth.setGasPriceStrategy(0)
+```
+
+### Step 4: Deploy Helper Contracts
+NOTE: In addition to the two helper contracts `msg_hasher` and `purity_checker` you also need to deploy the `rlp_decoder` Contract. On a production chain, an `rlp_decoder` contract is already deployed and vyper’s standard internal library knows it’s address and gives vyper contracts access to some functionality at that address.
+**When using a test chain, these helper contract must be deployed before the casper contract.**
```

Comment:
Important to note that `rlp_decoder` must be deployed before _compiling_ the casper contract.